classifier_and_summarizer/summarizer/text_media_matcher.py

```diff
@@ -0,0 +1,199 @@
+        self.media_preference_for_sentence = media_preference_for_sentence
+        self.sentence_preference_for_media = sentence_preference_for_media
+
+    def get_matching(self):
+        return self.__gale_shapley_matching()
+
+    def __media_rank_in_sentence_preference(self, sentence_index, media_index):
+        return self.media_preference_for_sentence[sentence_index].index(
+            media_index)
+
+    def __sentence_has_better_preference(
+            self, sentence_index, unmatched_index):
+        return self.__media_rank_in_sentence_preference(
+            sentence_index, unmatched_index) < \
+            self.__media_rank_in_sentence_preference(
+            sentence_index, self.media_matched_for_sentence[sentence_index])
+
+    def __gale_shapley_matching(self):
+        '''
+        Finds the stable matching between the text and media
+        Given two (n,n) matrices of preferences for the set of Sentences, Media
+        finds the stable matching by running the gale-shapley
+        matching algorithm
+        Returns : a list of tuples where each tuple (x,y) means
+        x = index of sentence
+        y = index of media
+        Thus, it returns the indices matched as a list of tuples
+        '''
+        # Make the matching optimal for the Media
+        some_media_is_unmatched = True
+
+        # -1 denotes it is currently unmatched
+        self.sentence_matched_for_media = [-1] * self.set_size
+        self.media_matched_for_sentence = [-1] * self.set_size
+
+        self.count_of_unmatched_media = self.set_size
+
+        while some_media_is_unmatched:
+            unmatched_media_index = -1  # no index found currently
+            for i in range(self.set_size):
+                if self.sentence_matched_for_media[i] == -1:
+                    unmatched_media_index = i
+                    break
+
+            for i in self.sentence_preference_for_media[unmatched_media_index]:
+                # the sentence is unmatched
+                if self.media_matched_for_sentence[i] == -1:
+                    # we can match the sentence directly
+                    self.media_matched_for_sentence[i] = unmatched_media_index
+                    self.sentence_matched_for_media[unmatched_media_index] = i
+                    self.count_of_unmatched_media -= 1
+                    break
+                elif self.__sentence_has_better_preference(
+                        i,
+                        unmatched_media_index):
+                    # i prefers the current media better
+                    # unmatch media currently matched for sentence i
+                    self.sentence_matched_for_media[
+                        self.media_matched_for_sentence[i]] = -1
+                    self.sentence_matched_for_media[unmatched_media_index] = i
+                    self.media_matched_for_sentence[i] = unmatched_media_index
+                    break
+
+            some_media_is_unmatched = (self.count_of_unmatched_media > 0)
```

Comment:
Do we need this boolean now? We can directly use self.count_of_unmatched_media in the while loop.

Reply:
Yes, but the it was originally a boolean - and this would make it an integer - is this okay ?

Reply:
You are right that maintaining it as a boolean would be better and in accordance with its name. What I meant was that the variable "some_media_is_unmatched" itself might be not required and redundant now that we have "self.count_of_unmatched_media". Do correct me if I am wrong here though.

Reply:
That makes sense ! I interpreted it wrongly. Will update accordingly.